src/explain.ts

```diff
@@ -0,0 +1,78 @@
+import type { Document } from '.';
+import type { ExplainOptions } from './operations/explainable_command';
+import type { Server } from './sdam/server';
+import { maxWireVersion } from './utils';
+
+/** @public */
+export enum Verbosity {
+  queryPlanner = 'queryPlanner',
+  queryPlannerExtended = 'queryPlannerExtended',
+  executionStats = 'executionStats',
+  allPlansExecution = 'allPlansExecution'
+}
+
+/**
+ * For backwards compatibility, true is interpreted as
+ * "allPlansExecution" and false as "queryPlanner".
+ * @public
+ */
+export type VerbosityLike = Verbosity | boolean;
+
+// Minimum server versions which support explain with specific operations
+const SUPPORTS_EXPLAIN_WITH_REMOVE = 3;
+const SUPPORTS_EXPLAIN_WITH_UPDATE = 3;
+const SUPPORTS_EXPLAIN_WITH_DISTINCT = 3.2;
+const SUPPORTS_EXPLAIN_WITH_FIND_AND_MODIFY = 3.2;
+const SUPPORTS_EXPLAIN_WITH_MAP_REDUCE = 4.4;
+
+/** @internal */
+export class Explain {
+  verbosity: Verbosity;
+
+  constructor(verbosity: VerbosityLike) {
+    if (typeof verbosity === 'boolean') {
+      this.verbosity = verbosity ? Verbosity.allPlansExecution : Verbosity.queryPlanner;
+    } else {
+      this.verbosity = Verbosity[verbosity];
+    }
+  }
+
+  static fromOptions(options?: ExplainOptions): Explain | undefined {
+    if (options == null || options.explain === undefined) {
+      return;
+    }
+    return new Explain(options.explain);
+  }
+
+  static valid(options?: ExplainOptions): boolean {
+    if (options == null || options.explain === undefined) {
```

Comment:
```suggestion
    if (!options?.explain) {
```
I think this should do the trick, if options is nullish then `!` makes this true, if explain is nullish then its still true. 
Was it intentional to explicitly check for options.explain being undefined vs `==` double equal nullish?

Reply:
Not intentional. Just wanted to avoid missing cases where `options.explain = false`, since we do actually want to explain the command in that case (confusing, I know), so I was checking for `=== undefined` whenever accessing explain from options.